hasher-matcher-actioner/terraform/main.tf

```diff
@@ -284,6 +284,47 @@ resource "aws_s3_bucket" "data_bucket" {
   force_destroy = true
 }
 
+/**
+ * # Banks S3 Media Bucket:
+ * Stores media uploaded by bank admins. Store raw data. Should ideally be
+ * closed to the wider internet and can have retention rules based on legal
+ * policy.
+ */
+resource "aws_s3_bucket" "banks_media_bucket" {
+  bucket_prefix = "${var.prefix}-banks-media-"
+  acl = "private"
+  tags = merge(
+    var.additional_tags,
+    {
+      Name = "BanksMediaBucket"
+    }
+  )
+
+  versioning {
+    # Don't see a reason for versioning media. Everything will be keyed on uuid
+    # and current date. 
+    enabled = false
+  }
+
+  cors_rule {
+    # The webapp uploads media to this bucket directly.
+    allowed_headers = ["*"]
+    allowed_methods = ["PUT"]
+
+    # Partners could lock this down to specific origin where possible.
+    allowed_origins = ["*"]
+    max_age_seconds = 3000
+  }
+
+  lifecycle {
+    # To prevent execution of plans which would cause this bucket to get
+    # destroyed. Once in the hands of partners, we have to be extra careful to
+    # not accidentally delete their data. 
+    prevent_destroy = true
```

Comment:
I've see you add before, I don't disagree with the motives but it makes teardown to build testing harder can you make it a tf.vars arg that defaults to true that way I can configure "living dangerously" in one place when testing destroy and fresh apply

Reply:
Aah. That is a valid point. I never face this because I _never_ teardown. But I can see what you mean.

Defer to another PR? #804 to track.